src/+datools/+error/Gaussian.m

```diff
@@ -1,27 +1,43 @@
-classdef Gaussian < datools.error.Error
+classdef Gaussian < datools.error.Uncertainty
+    %GAUSSIAN Gaussian child class for adding gaussina noise with a user
+    %   defined deviation and bias
 
     properties
-        CovarianceSqrt
-        Bias
+        Covariance % Covariance for the noise
+        Bias % Bias
+    end
+
+    properties (Dependent)
+        CovarianceSqrt % Covariance square root
     end
 
     methods
 
         function obj = Gaussian(varargin)
             p = inputParser;
-            addOptional(p, 'CovarianceSqrt', 1);
+            addOptional(p, 'Covariance', 1);
             addOptional(p, 'Bias', 0);
             parse(p, varargin{:});
 
             s = p.Results;
 
-            obj.CovarianceSqrt = s.CovarianceSqrt;
+            obj.Covariance = s.Covariance;
             obj.Bias = s.Bias;
         end
 
-        function xp = adderr(obj, ~, x)
-            A = obj.CovarianceSqrt * randn(size(x, 1), size(x, 2));
-            xp = x + A + obj.Bias;
+
+        function error = sample(obj, ~, x)
```

Comment:
This should not take x at all now. This should just be N.

Reply:
We don't need like. most of our algorithms are useless with VPA and most won't work with dlarray, so just don't